src/components/tabs/helpers/tab-nesting-context.tsx

```diff
@@ -0,0 +1,21 @@
+import { createContext, useContext, ReactNode } from 'react'
+
+type NestedTabContextValue = false
+
+export function useIsNested(): NestedTabContextValue {
+  return useContext(NestedTabContext)
+}
+
+const NestedTabContext = createContext(undefined)
```

Comment:
Shoutout Bryce for coming up with the idea to use `Context` while we were working on #495 🙌